lib/sigh/resign.rb

```diff
@@ -112,19 +120,63 @@ def ask_for_signing_identity
       ask('Signing Identity: ')
     end
 
-    # Array of available signing identities
-    def installed_identies
-      available = `security find-identity -v -p codesigning`
-      ids = []
+    # Hash of available signing identities
+    def installed_identities
+      available = `security find-identity -p codesigning`
+      ids = {}
       available.split("\n").each do |current|
         begin
-          (ids << current.match(/.*\"(.*)\"/)[1])
+          match = current.match(/.*([0-9A-Z]{40}) \"(.*)\"(\s+\((.*)\))?/)
+          sha1 = match[1]
+          name = match[2]
+          issue = match[4]
+          ids[sha1] = SigningIdentity.new(sha1, name, issue)
         rescue
           nil
         end # the last line does not match
       end
 
-      ids
+      ids.values.select(&:may_be_valid?).sort_by do |identity|
+        [identity.name, identity.valid? ? 0 : 1]
+      end
+    end
+
+    def installed_identity_descriptions
+      descriptions = []
+      installed_identities.group_by(&:name).each do |name, identities|
+        descriptions << (identities.any?(&:valid?) ? name.black : name.white)
```

Comment:
I think that using `.black` and `.white` here is a little concerning. Do you know how this behaves on terminals which are either black or white backgrounds (which I think are very common)? Could you provide some screenshots?